analytics.go

```diff
@@ -165,53 +171,120 @@ func (r *RedisAnalyticsHandler) Init() {
 
 	analytics.Store.Connect()
 
-	ps := config.Global().AnalyticsConfig.PoolSize
+	ps := r.globalConf.AnalyticsConfig.PoolSize
 	if ps == 0 {
 		ps = 50
 	}
+	log.WithField("ps", ps).Debug("Analytics pool workers number")
 
-	r.AnalyticsPool, err = tunny.CreatePoolGeneric(ps).Open()
-	if err != nil {
-		log.Error("Failed to init analytics pool")
+	r.recordsBufferSize = r.globalConf.AnalyticsConfig.RecordsBufferSize
+	if r.recordsBufferSize < minRecordsBufferSize {
+		r.recordsBufferSize = minRecordsBufferSize // force it to this value
```

Comment:
I think it is a bit unfair to call this variable `recordsBufferSize`, because its not total value, actual buffer will be `pool * buffer_size`. It feels like doing `recordsBufferSize = recordsBufferSize / pool_size` will be more fair.

Reply:
I am bad in naming things, will come up with better name

Reply:
Or you can leave it as it is, like "buffer_size", and per pool buffer will be: BufferSize / pool_size. 

IMHO, it is the most logical here. You separate control concurrency via pool size and total buffer size. Just set it to bigger value if you want, like 10000. 

For sure if we will be able to auto-grow this buffer, and we do not need to expose this option at all, it will be even more cooler, but only if implementation does not take too much time.